src/utill.rs

```diff
@@ -292,7 +284,6 @@ pub fn redeemscript_to_scriptpubkey(redeemscript: &ScriptBuf) -> ScriptBuf {
         &redeemscript.wscript_hash().to_byte_array(),
     )
     .unwrap();
```

Comment:
Damn, this opened up an entire can of worms. This is such a low-level util that changing its signature affects everything. But this needed to be handled or it can be quite problematic. A rouge maker can trigger this unwrap by sending non-segwit reedemscripts. 

Stay tuned for another round of refactoring commits.

Reply:
its gonna be a mess later too trying to fix it. Pushed the changes. lets bite the bullet here.